pkg/service/orchestrator/rerecord.go

```diff
@@ -0,0 +1,252 @@
+				return nil
+			})
+		}
+
+		var err error
+		select {
+		case err = <-errCh:
+			if err != nil {
+				stopReason = "error while starting the recording"
+				utils.LogError(o.logger, err, stopReason, zap.String("testset", testSet))
+			}
+		case err = <-replayErrCh:
+			if err != nil {
+				stopReason = "error while replaying the testcases"
+				utils.LogError(o.logger, err, stopReason, zap.String("testset", testSet))
+			}
+		case <-ctx.Done():
+		}
+
+		if err == nil || ctx.Err() == nil {
+			// Sleep for 3 seconds to ensure that the recording has completed
+			time.Sleep(3 * time.Second)
+		}
+
+		recordCtxCancel()
+
+		// Wait for the recording to stop
+		err = errGrp.Wait()
+		if err != nil {
+			utils.LogError(o.logger, err, "failed to stop re-recording")
+		}
+
+		// Check if the global context is done after each iteration
+		if ctx.Err() != nil {
+			break
+		}
+	}
+
+	if stopReason == "" {
+		if ctx.Err() != nil {
+			stopReason = "context cancelled"
+			o.logger.Warn("Re-record was cancelled, keploy might have not recorded few test cases")
+			return nil
+		}
+		stopReason = "Re-recorded all the selected testsets successfully"
+		o.logger.Info("Re-record was successfull. Do you want to remove the older testsets?", zap.Any("testsets", SelectedTests))
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			o.logger.Warn("Failed to read input. The older testsets will be kept.")
+			return nil
+		}
+
+		if input == "y\n" || input == "Y\n" {
+			for _, testSet := range SelectedTests {
+				err := o.replay.DeleteTestSet(ctx, testSet)
+				if err != nil {
+					o.logger.Warn("Failed to delete the testset", zap.String("testset", testSet))
+				}
+			}
+			o.logger.Info("Deleted the older testsets successfully")
+		} else if input == "n\n" || input == "N\n" {
+			o.logger.Info("skipping the deletion of older testsets")
+		} else {
+			o.logger.Warn("Invalid input. The older testsets will be kept.")
+		}
+		return nil
+	}
+
+	utils.LogError(o.logger, err, stopReason)
+	return fmt.Errorf(stopReason)
+}
+
+func (o *Orchestrator) replayTests(ctx context.Context, testSet string) (bool, error) {
+	var err error
+	var errMsg string
+
+	//replay the recorded testcases
+
+	tcs, err := o.replay.GetTestCases(ctx, testSet)
+	if err != nil {
+		errMsg = "Failed to get all testcases"
+		utils.LogError(o.logger, err, errMsg, zap.String("testset", testSet))
+		return false, fmt.Errorf(errMsg)
+	}
+
+	if len(tcs) == 0 {
+		o.logger.Warn("No testcases found for the given testset", zap.String("testset", testSet))
+		return false, nil
+	}
+
+	host, port, err := pkg.ExtractHostAndPort(tcs[0].Curl)
+	if err != nil {
+		errMsg = "failed to extract host and port"
+		utils.LogError(o.logger, err, "")
+		o.logger.Debug("", zap.String("curl", tcs[0].Curl))
+		return false, fmt.Errorf(errMsg)
+	}
+	cmdType := utils.CmdType(o.config.CommandType)
+
+	var userIP string
+	if utils.IsDockerKind(cmdType) {
```

Comment:
maybe we should rename `IsDockerKind` to `isDockerCmd`

Reply:
sure